Foundation/ConsistentCasting.swift

```diff
@@ -0,0 +1,272 @@
+ :--------+----+-----+-----+------+-----+------+-----+------+-----+------+----+---+----+-------+--------:
+ |Int32   |    |     |     |      |DL   |      |     |      |     |      |    |   |    |       |D       |
+ :--------+----+-----+-----+------+-----+------+-----+------+-----+------+----+---+----+-------+--------:
+ |UInt32  |    |     |     |      |     |DL    |     |      |     |      |    |   |    |       |D       |
+ :--------+----+-----+-----+------+-----+------+-----+------+-----+------+----+---+----+-------+--------:
+ |Int64   |    |     |     |      |     |      |DL   |      |     |      |    |   |    |       |D       |
+ :--------+----+-----+-----+------+-----+------+-----+------+-----+------+----+---+----+-------+--------:
+ |UInt64  |    |     |     |      |     |      |     |DL    |     |      |    |   |    |       |D       |
+ :--------+----+-----+-----+------+-----+------+-----+------+-----+------+----+---+----+-------+--------:
+ |Float   |    |     |     |      |     |      |     |      |DL   |      |    |   |    |       |D       |
+ :--------+----+-----+-----+------+-----+------+-----+------+-----+------+----+---+----+-------+--------:
+ |Double  |    |     |     |      |     |      |     |      |     |DL    |    |   |    |       |D       |
+ :--------+----+-----+-----+------+-----+------+-----+------+-----+------+----+---+----+-------+--------:
+ |Bool    |    |     |     |      |     |      |     |      |     |      |DL  |   |    |       |D       |
+ :--------+----+-----+-----+------+-----+------+-----+------+-----+------+----+---+----+-------+--------:
+ |Int     |    |     |     |      |     |      |     |      |     |      |    |DL |    |       |D       |
+ :--------+----+-----+-----+------+-----+------+-----+------+-----+------+----+---+----+-------+--------:
+ |UInt    |    |     |     |      |     |      |     |      |     |      |    |   |DL  |       |D       |
+ :--------+----+-----+-----+------+-----+------+-----+------+-----+------+----+---+----+-------+--------:
+ |Decimal |    |     |     |      |     |      |     |      |     |      |    |   |    |DL     |D       |
+ :--------+----+-----+-----+------+-----+------+-----+------+-----+------+----+---+----+-------+--------:
+ |NSNumber|D   |D    |D    |D     |D    |D     |D    |D     |D    |D     |D   |D  |D   |       |DL      |
+ :--------+----+-----+-----+------+-----+------+-----+------+-----+------+----+---+----+-------+--------:
+ ```
+
+ - Returns: Casted representation of passed object, or `nil` if casting failed.
+
+ - Parameter object: Object to cast.
+ */
+func platformConsistentCast<T>(_ object: Any) -> T? {
+    if let casted = object as? T {
+        return casted
+    }
+
+    if let safeBridgeable = object as? _PlatformConsistentCasting {
+        return safeBridgeable._casted()
+    }
+
+    return nil
+}
+
+protocol _PlatformConsistentCasting {
```

Comment:
we have had issues where declarations accidentally leak out beyond the module of Foundation (however I haven't seen that since).

Even though internal is the default application of scope, it just seems to be good hygiene to make certain they are marked as such.